lib/p2p/packets/types/DisconnectingPacket.ts

```diff
@@ -15,6 +18,46 @@ class DisconnectingPacket extends Packet<DisconnectingPacketBody> {
   public get direction() {
     return PacketDirection.Unilateral;
   }
+
+  public static deserialize(binary: Uint8Array): DisconnectingPacket | undefined {
+    const msg = pb.DisconnectingPacket.deserializeBinary(binary).toObject();
+    return DisconnectingPacket.validate(msg) ? DisconnectingPacket.convert(msg) : undefined;
+  }
+
+  private static validate(msg: pb.DisconnectingPacket.AsObject): boolean {
+    return !!(msg.header
+      && msg.header.id
+      && msg.header.hash
+      && !msg.header.reqid
+      && msg.reason
+    );
+  }
+
+  private static convert = (msg: pb.DisconnectingPacket.AsObject): DisconnectingPacket => {
+    return new DisconnectingPacket({
+      header: {
+        id: msg.header!.id,
+        hash: msg.header!.hash,
+      },
+      body: removeUndefinedProps({
```

Comment:
What's the point of using `removeUndefinedProps` here for example? Isn't it fine if `payload` is undefined? I haven't examined the other usages of it but I'm not sure if it's needed.

Reply:
It's basically the same problem we previously had - an object with an undefined prop isn't the same as an object without that prop. Because of that, the objects deep equal assertions will fail.
You can remove the `removeUndefinedProps` and see yourself how that tests are failing.

It's not a big deal, and we can relax the test assertions,  but I think it's still better to remove the prop entirely.